aten/src/ATen/autocast_mode.h

```diff
@@ -21,7 +21,7 @@ inline at::ScalarType prioritize(at::ScalarType current, const Tensor& nextArg)
     AT_ERROR("promote type is double in at::autocast::prioritize");
     return current;
   }
-  if (nextArg.is_cuda() && nextArg.is_floating_point()) {
+  if ((nextArg.is_cuda() || nextArg.is_xla()) && nextArg.is_floating_point()) {
```

Comment:
It would be better to centralize these tests into a single function, `is_autocast_eligible(const Tensor&)` which can do this test.

Reply:
Done.